src/steamship/agents/react/react.py

```diff
@@ -79,6 +82,11 @@ def next_action(self, context: AgentContext) -> Action:
             chat_history=self.messages_to_prompt_history(
                 messages=context.chat_history.select_messages(self.message_selector)
             ),
+            relevant_history=self.messages_to_prompt_history(
+                context.chat_history.search(context.chat_history.last_user_message.text, k=3)
+                .wait()
```

Comment:
what happens if `wait()` fails?  should we try this in a recoverable fashion?

Reply:
I'm honestly not sure why `search` is a task to begin with.  Searching is not a class of operation I expect to be asynchronous.